src/adapter/src/coord.rs

```diff
@@ -1139,8 +1144,17 @@ pub async fn serve<S: Append + 'static>(
                 segment_client,
                 metrics: Metrics::register_with(&inner_metrics_registry),
             };
-            let bootstrap =
-                handle.block_on(coord.bootstrap(builtin_migration_metadata, builtin_table_updates));
+            let bootstrap = handle.block_on(async {
+                coord
+                    .bootstrap(builtin_migration_metadata, builtin_table_updates)
```

Comment:
It's possible that bootstrapping creates new objects with a user ID higher than `next_ids.0` due to the builtin schema migration.

Reply:
You can probably fix this by getting the next IDs after bootrapping.

Reply:
I think it's fine to get those before the migration, I'm interested in objects that would be left over from a previous environmentd run, so no need to look at the ID's being allocated during bootstrap and normal operation.

The only thing I'm worried about is that someone else does a  `create replica` and we see the increased replica ID in the `next_replica_id`, but not the replica object itself. Then `remove_orphans` would gladly remove this from the orchestrator. Afaik this can not happen, because both the initial catalog load and replica create happen inside a transaction.

Reply:
If we see a new object created during migration, when removing orphans then will this code panic in the storage controller?
```
            if id >= next_id {
                // Found a storaged in kubernetes with a higher id than what we are aware of. This
                // must have been created by an environmentd with a higher epoch number.
                panic!(
                    "Found storaged id ({}) in orchestrator >= than next_id ({})",
                    id, next_id
                );
            }
```

Reply:
> The only thing I'm worried about is that someone else does a create replica and we see the increased replica ID in the next_replica_id, but not the replica object itself. Then remove_orphans would gladly remove this from the orchestrator. Afaik this can not happen, because both the initial catalog load and replica create happen inside a transaction.

I think we're OK here. A user can't create a replica until we're done with bootstrapping and finished removing orphaned nodes. If a new Coordinator creates a new replica and increases the `next_replica_id` then our read will fail because we're no longer the leader.

Reply:
> If we see a new object created during migration, when removing orphans then will this code panic in the storage controller?
>

You're right, it won't be deleted, but we will panic: We need to fetch the `next_id` after migration. When I just fetch them after the bootstrap (using `peek_key_one` on the ID allocation collection), that call does an epoch check right?

Reply:
Yep. All stash reads and writes will do an epoch check.